### DISCUSSION TITLE
Store filled only if page refresh

### DISCUSSION THREAD

**Discussion by popuguytheparrot** [vercel/next.js#45711] [Help] [answered]
Summary
Hi. I use effector and try hydrate state on ssr.

When refresh ssr page in browser, initialState has data, but if use LInk, no initialState

Example
https://stackblitz.com/edit/nextjs-l381hk

**Comment by icyJoseph** [comment, marked as answer]
Hi,

I think the issue is here:

```js
let scope;

if (typeof window !== 'undefined') {
  console.info('scope', scope);
}

function initializeScope(initialData) {
  console.log('initialData', initialData);

  let _scope =
    scope ??
    fork({
      values: initialData,
    });

  // For SSG and SSR always create a new scope
  if (typeof window === 'undefined') return _scope;
  // Create the scope once in the client
  if (!scope) scope = _scope;

  console.log(scope);

  return _scope;
}
```

Because you are assuming that a page that runs GSSP is always fully generated Server Side, but that's not really what Next.js does.

## User lands on your GSSP page

A user navigates directly to your page (or refreshes the page), then GSSP starts, the result is used to render an HTML page, and this result is sent to the client. A whole HTML page, with scripts, CSS, etc, is made.

## SPA transition into your GSSP page

A user navigates to your GSSP page use `next/link`. As soon as the `next/link` mounts, Next.js fetches the JavaScript that can generate this page. 

When the user transitions in to the page using `next/link`, instead of unloading the current page, and causing more delays, since Next.js already has the JavaScript for the page, it only needs the latest JSON that drives the page, in order to produce a full page. 

So it makes a request for the JSON only, and uses that JSON as props to render the page as Client Side rendering. Do understand that GSSP runs Server Side Only, only the JSON response is sent from the Server to the Client.

People always miss this detail when it comes to GSSP, although the docs are very clear https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props#when-does-getserversideprops-run

So in your `useScope` code you always ignore the newly incoming `initialData`, which is happening client side. 

Do you follow my explanation? I can understand that GSSP is a bit hard to grasp.

**Reply by popuguytheparrot** [reply]
Thanks for the answer. I didn’t understand about the moment with JSON. If need to always have new data when navigate to ssr page, then how to implement this? Because in the console I see that the initialDate doesn't come at all on the spa transition. Only if refresh page by F5

**Reply by icyJoseph** [reply]
You are getting the data, but it is not logging on the terminal because that's Server Side, and as I said, this is happening Client Side, because of the SPA transition, see this video where initial data contains `1337`.

https://user-images.githubusercontent.com/21013447/217803066-0f34848f-2b2b-4a56-9532-fc4094a076b3.mov

The problem is the client you check that scope exists, and if it exists, then you skip the update, and ignore the newly incoming data.

https://user-images.githubusercontent.com/21013447/217803825-d1b36fe1-d8e9-43ec-924e-a47087e8ff99.mov

However, I think that introduces some issues, if I understand this data management library correctly.

**Reply by popuguytheparrot** [reply]
Understood 👍
